Polish language updates + RFCOMM mode implementation

WinLin97 opened the pull request:
continuation of [topic](https://github.com/Fabi019/hid-barcode-scanner/issues/301)

including:

  Polish Language Updates:
  - Updated Polish keyboard layout
  - Updated Polish strings.xml translations

  RFCOMM Mode Implementation:
  - Introduced RFCOMM Bluetooth connection mode
  - Added RFCOMM connection improvements
  - Implemented AutoConnect functionality for RFCOMM
  - Multiple RFCOMM mode enhancements

  Additional Changes:
  - Scanner.kt updates to align with latest basis
  - TemplateProcessor refactoring
  - Various code polish and improvements

Fabi019 commented on app/src/main/java/dev/fabik/bluetoothhid/bt/BluetoothService.kt:
```diff
@@ -53,6 +53,15 @@ class BluetoothService : Service() {
         CoroutineScope(Dispatchers.IO).launch {
             controller?.register()
         }
+
+        // Start observing connection mode changes for auto RFCOMM switching
+        controller?.startModeObserver()
+
+        // Start observing auto-connect changes
+        controller?.startAutoConnectObserver()
+
+        // Start Bluetooth state monitoring
+        controller?.startBluetoothStateMonitoring()
```
I would put these into the register method otherwise when unregister is called the monitoring might not be started anymore when only register is called (Also the register method could would not need to be suspend anymore since it can the use the autoConnect from the observer)

WinLin97 replied:
✅ FIXED
  If unregister() is called followed by only register() (without service restart), the
  observers wouldn't restart since they were only initialized in onCreate(). Moved startModeObserver(),
  startAutoConnectObserver(), and startBluetoothStateMonitoring() to the register() method for proper lifecycle
  management.

Fabi019 commented on app/src/main/java/dev/fabik/bluetoothhid/bt/RfcommController.kt:
```diff
@@ -0,0 +1,320 @@
+
+    fun connectRFCOMM(){
+        controllerScope.launch {
+            connectionMode = context.getPreference(PreferenceStore.CONNECTION_MODE).first()
```
The connectRFCOMM should be only called when in the right mode, so this additional check should not be neccessary

Fabi019 commented on app/src/main/java/dev/fabik/bluetoothhid/bt/BluetoothController.kt:
```diff
@@ -155,6 +184,95 @@ class BluetoothController(var context: Context) {
+                        }
+                        else -> {
+                            Log.d(TAG, "Switching to HID mode - disconnecting RFCOMM")
+                            rfcommController.disconnectRFCOMM()
```
I think it could make sense to put all the HID specific registration/unregistration into a own registerHid/unregisterHid method (still also called from the main register/unregister) that could be then called here when in the right mode

WinLin97 replied:
✅ FIXED - Absolutely correct! This was mine logical error. In RFCOMM mode, we shouldn't attempt HID connections
  since RFCOMM operates as a server waiting for connections, not initiating them. Modified connect() method to check
   connection mode first and only update currentDevice for UI purposes in RFCOMM mode.

Fabi019 commented on app/src/main/java/dev/fabik/bluetoothhid/bt/BluetoothController.kt:
```diff
@@ -155,6 +184,95 @@ class BluetoothController(var context: Context) {
+                .distinctUntilChanged()
+                .debounce(200)
+                .collect { mode ->
+                    Log.d(TAG, "Connection mode changed to: $mode")
```
Since we are already listening to changes of the connection mode the current value could be stored as a class variable and used everywhere else in this file instead of always requiring to check them mode with `context.getPreference(PreferenceStore.CONNECTION_MODE).first()`

WinLin97 replied:
✅ FIXED

Fabi019 commented on app/src/main/java/dev/fabik/bluetoothhid/bt/BluetoothController.kt:
```diff
@@ -218,6 +351,9 @@ class BluetoothController(var context: Context) {
         Log.d(TAG, "connecting to $device")
 
+        // Set expected device for RFCOMM filtering
+        rfcommController.setExpectedDeviceAddress(device.address)
```
When in rfcomm mode everything following should not be needed. It will be neccessary however to update the currentDevice with `hostDevice.update { device }` to simulate a connected device.

Fabi019 commented on app/src/main/java/dev/fabik/bluetoothhid/Scanner.kt:
```diff
@@ -545,9 +545,15 @@ fun ToggleFlashButton(camera: CameraControl?, info: CameraInfo) {
 fun BoxScope.CapsLockWarning() {
```
I think it would be good to notify the user when the target device is connected to us. Maybe something simmilar to the current "No device connected" warning should indicate like "Listening for new connection from ...". That way the user would also know that the connection needs to be initiated by the PC

Fabi019 commented on app/src/main/java/dev/fabik/bluetoothhid/bt/BluetoothController.kt:
```diff
@@ -88,6 +114,9 @@ class BluetoothController(var context: Context) {
             Log.d(TAG, "onServiceDisconnected")
 
+            // Disconnect RFCOMM
+            rfcommController.disconnectRFCOMM()
```
Same as L88 I would remove this from the service listener as it is associated with the HID connection

Fabi019 reviewed (commented):
I added some feedback 😀 For me I first needed to add a new outgoing COM port in the bluetooth settings as it was not created automatically. Was this also the case for you?

WinLin97 commented on app/src/main/java/dev/fabik/bluetoothhid/bt/BluetoothController.kt:
```diff
@@ -59,10 +72,23 @@ class BluetoothController(var context: Context) {
         override fun onServiceConnected(profile: Int, proxy: BluetoothProfile) {
             Log.d(TAG, "onServiceConnected")
 
+            // Initialize RFCOMM if needed
+            MainScope().launch {
+                val connectionMode = context.getPreference(PreferenceStore.CONNECTION_MODE).first()
+                if (connectionMode == 1) {
+                    rfcommController.connectRFCOMM()
+                }
+            }
```
✅ FIXED - You are absolutely correct. The startModeObserver() already handles RFCOMM connection/disconnection
  based on mode changes, and since it's called in service creation, it immediately gets the current value via
  .collect{}. The duplicate calls in onServiceConnected and onServiceDisconnected have been removed.

WinLin97 commented:
> For me I first needed to add a new outgoing COM port in the bluetooth settings as it was not created automatically. Was this also the case for you?

I don't remember - I have this setted in Windows and I won't deleted it since 11 months... 
Maybe it's not setting automatically?
Let You know if there is some way to do this automatically 👍 but probably this cause manual action...

I've done the code refining - if You have some other annotations, please let me know.

WinLin97 commented on app/src/main/java/dev/fabik/bluetoothhid/bt/RfcommController.kt:
```diff
@@ -0,0 +1,336 @@
+        L("Expected device address set to: ${addr ?: "any"}")
+    }
+
+    fun setAutoConnectEnabled(enabled: Boolean) {
```
Comment included

Fabi019 commented:
> Maybe the better way is to move "Allow insecure RFCOMM" under "Developer Mode" in the menu on top right?
I think that this would be the better option as the setting is more like a workaround that the normal user should not need. But otherwise everything looks good to me now 👍

WinLin97 commented:
I saw the Dropdown menu includes all the functions, that are additional - not every user needs that.
So i moved the "Insecure RFCOMM" here, and added more describing warn - just like You've in OCR.
Something else You need to change? 😀
